File: Part 2 - Regression/Section 4 - Simple Linear Regression/my_regression.R
# Data Preprocessing Template

# Importing the dataset
dataset = read.csv('Salary_Data.csv')
#dataset = dataset[,2:3]

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)


# Feature Scaling
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])
regressor = lm(formula = Salary ~ YearsExperience, training_set)

y_pred = predict(regressor, newdata = test_set)

#install.packages('ggplot2')
library(ggplot2)
ggplot() + 
  geom_point(aes(x = training_set$YearsExperience, y =training_set$Salary),
             colour = 'red') + 
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            colour = 'blue') + 
  ggtitle('Salary vs exp (training)') +
  xlab('years of exp')+
  ylab('Salary')

ggplot() + 
  geom_point(aes(x = test_set$YearsExperience, y =test_set$Salary),
             colour = 'red') + 
  geom_line(aes(x = test_set$YearsExperience, y = predict(regressor, newdata = test_set)),
            colour = 'blue') + 
  ggtitle('Salary vs exp (test)') +
  xlab('years of exp')+
  ylab('Salary')

File: Part 2 - Regression/Section 4 - Simple Linear Regression/my_regression.py
# -*- coding: utf-8 -*-
"""
Created on Sun Sep  8 23:16:25 2019

@author: Sowmitha
"""

# -*- coding: utf-8 -*-
"""
Created on Sun Sep  8 17:39:16 2019

@author: Sowmitha
"""

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

#from sklearn.preprocessing import Imputer
#imputer = Imputer(missing_values = "NaN", strategy = "mean", axis = 0)
#imputer = imputer.fit(X[:, 1:3])
#X[:, 1:3] = imputer.transform(X[:, 1:3])

#from sklearn.preprocessing import LabelEncoder, OneHotEncoder
#labelencoder_X = LabelEncoder()
#X[:, 0] = labelencoder_X.fit_transform(X[:, 0])
#onehotencoder = OneHotEncoder(categorical_features = [0])
#X = onehotencoder.fit_transform(X).toarray()

#labelencoder_y = LabelEncoder()
#y = labelencoder_y.fit_transform(y)

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test =  train_test_split(X, y, test_size = 1/3, 
                                                     random_state = 0)
"""
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)"""

from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

y_pred = regressor.predict(X_test)

plt.scatter(X_train, y_train, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color='blue')
plt.title('salary Vs exp (training)')
plt.xlabel('years of exp')
plt.ylabel('salary')
plt.show()


plt.scatter(X_test, y_test, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color='blue')
plt.title('salary Vs exp (test)')
plt.xlabel('years of exp')
plt.ylabel('salary')
plt.show()


